### ISSUE TITLE
assetDirectory has 2 meaning

### ISSUE THREAD

**Issue by kireerik** [kireerik/refo#7]
Default usage:
https://github.com/kireerik/refo/blob/c9b131a1f13dcac3d322c995051b936578e6d645/index/index/core/watch/index.js#L28
Required because of inlined sources (svg, css). [inline-source](https://github.com/popeindustries/inline-source)

&nbsp;

`asset` directory within `siteDirectory`:
https://github.com/kireerik/refo/blob/c9b131a1f13dcac3d322c995051b936578e6d645/index/index/main/handlers.js#L42
Required because of front end JavaScript handling.

&nbsp;

This could be a feature as well, but it can be also confusing. So it is probably recommended to separate the 2 use cases and have a different `assetDirectory` variable for front end JavaScript handling.

**Comment by kireerik** [comment]
Alternative way to handle front end JavaScript:

Main handler:
https://github.com/kireerik/refo/blob/ec7f6c3a7e33339f9249f2c6707bf5f161ee6ea1/index/index/main/handlers.js#L61-L64

```JavaScript
if (typeof html == 'string') {
	html = await handleHTML(filePath, html)

	;(filePath.substring(0, filePath.lastIndexOf('.')) + '.html')
		.saveToStaticDirectory(html)
} else
	filePath.saveToStaticDirectory(html())
```


Front end JavaScript:
```JavaScript
const handleJs = require('frontend-javascript-handler')

module.exports = () => handleJs('../frontEndJavaScriptFileToHandle')
```

Front end JavaScript handler module:
```JavaScript
const path = require('path')

, bundle = require('bundle-js')
, minifyJS = require('uglify-js').minify

module.exports = file => {
	const filePath = path.resolve(file)

	;(() => require(filePath))()

	let js = bundle({entry: filePath, disablebeautify: true})

	for (let i = 1; i <= 2; i++)
		js = minifyJS(js).code

	return js
}
```

This is not recommended because the `filePath` needs to be requred as a module even when it is not a module in order to have hot module replacement:
```JavaScript
	;(() => require(filePath))()
```

[Requiring it as a string](https://stackoverflow.com/questions/12752622/require-file-as-string/12753026#12753026) resolves this, but this is currently not supported by [hot-module-replacement](https://github.com/sidorares/hot-module-replacement):
- https://github.com/sidorares/hot-module-replacement/issues/6#issuecomment-445472448

**Comment by kireerik** [comment]
closed by #43
